src/System Application/App/Agent/Setup/AgentImpl.Codeunit.al

```diff
@@ -137,7 +138,29 @@ codeunit 4301 "Agent Impl."
         until AgentAccessControl.Next() = 0;
     end;
 
-    internal procedure SetProfile(AgentUserSecurityID: Guid; var AllProfile: Record "All Profile")
+    internal procedure PopulateProfile(ProfileID: Text[30]; ProfileAppID: Guid; var TempAllProfile: Record "All Profile" temporary)
+    begin
+        TempAllProfile.Init();
+        TempAllProfile.Scope := TempAllProfile.Scope::Tenant;
+        TempAllProfile."App ID" := ProfileAppID;
+        TempAllProfile."Profile ID" := ProfileID;
+        TempAllProfile.Insert();
+    end;
+
+    internal procedure AssignDefaultProfile(AgentUserSecurityID: Guid)
+    var
+        Agent: Record Agent;
+        TempAllProfile: Record "All Profile" temporary;
+        AgentFactory: Interface IAgentFactory;
+    begin
+        GetAgent(Agent, AgentUserSecurityID);
```

Comment:
It would be better to pass in the agent record, so we avoid the get

Reply:
Updated